models/address_book.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'entry'
+
+class AddressBook
+  attr_accessor :entries
+
+  def initialize
+    @entries = []
+  end
+
+  def add_entry(name, phone_number, email)
+    index = 0
+    @entries.each do |entry|
+      if name < entry.name
+        break
+      end
+      index += 1
+    end
+
+    @entries.insert(index, Entry.new(name, phone_number, email))
+  end
+# Code for assignment: remove_entry method.
+  def remove_entry(name, phone_number, email)
+    delete_entryj = nil
+
+    @entries.each do |entry|
+      if name == entry.name && phone_number == entry.phone_number && email == entry.email
+        @delete_entry = entry
```

Comment:
Right.  Is it needed outside of the method?

On Tue, Oct 27, 2015 at 1:58 PM Felipe Quintero <notifications@github.com>
wrote:

> In models/address_book.rb
> <https://github.com/ffquint/address_bloc/pull/1#discussion_r43160554>:
>
> > +    @entries.each do |entry|
> > +      if name < entry.name
>
>
> > +        break
> > +      end
> > +      index += 1
> > +    end
> > +
> > +    @entries.insert(index, Entry.new(name, phone_number, email))
> > +  end
> > +# Code for assignment: remove_entry method.
> > +  def remove_entry(name, phone_number, email)
> > +    delete_entryj = nil
> > +
> > +    @entries.each do |entry|
> > +      if name == entry.name && phone_number == entry.phone_number && email == entry.email
> > +        @delete_entry = entry
>
> Creating an instance variable means it will be available outside of the
> the method (remove_entry, in this case).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ffquint/address_bloc/pull/1/files#r43160554>.
>